src/main/java/com/zilen/weather/service/WeatherService.java

```diff
@@ -1,28 +1,40 @@
 package com.zilen.weather.service;
 
-import com.zilen.weather.entity.WeatherDTO;
+import com.zilen.weather.dto.Weather;
+import com.zilen.weather.exception.CityNotFoundException;
+import com.zilen.weather.repository.WeatherRepository;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.PropertySource;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 import org.springframework.web.client.RestTemplate;
 
+import java.util.Objects;
+
 @Service
 @PropertySource("classpath:application.properties")
 public class WeatherService {
 
     private final String appid;
     private final String url;
     private final RestTemplate restTemplate;
+    private final WeatherRepository weatherRepository;
+    private final WeatherConverter weatherConverter;
 
-    public WeatherService(@Value("${com.zilen.appid}") String appid, @Value("${com.zilen.url}") String url, RestTemplate restTemplate) {
+    public WeatherService(@Value("${com.zilen.appid}") String appid, @Value("${com.zilen.url}") String url, RestTemplate restTemplate, WeatherRepository weatherRepository, WeatherConverter weatherConverter) {
         this.appid = appid;
         this.url = url;
         this.restTemplate = restTemplate;
+        this.weatherRepository = weatherRepository;
+        this.weatherConverter = weatherConverter;
     }
 
-    public WeatherDTO findByCityName(String cityName) {
-        ResponseEntity<WeatherDTO> response = restTemplate.getForEntity(url + cityName + "&units=metric&appid=" + appid, WeatherDTO.class);
-        return response.getBody();
+    public Weather findByCityName(String cityName) {
+        if (!Objects.nonNull(cityName)) {
```

Comment:
I would be better to use `Objects.isNull(...)` intead of `!Objects.noNull(...)`, but as you use 11 java here you can use `cityName.isBlank()`.